scrunch/datasets.py

```diff
@@ -1017,24 +1021,32 @@ def change_owner(self, user=None, project=None):
         :param project: id, name or Project object
         :return:
         """
+        warn("Use Dataset.move() to move datasets between projects", DeprecationWarning)
         if user and project:
             raise AttributeError(
                 "Must provide user or project. Not both"
             )
-        owner_url = None
+
         if user:
+            warn("Changing owner to users is deprecated. Move to projects", DeprecationWarning)
             if not isinstance(user, User):
                 user = get_user(user)
             owner_url = user.url
-        if project:
+            self.resource.patch({'owner': owner_url})
+            self.resource.refresh()
+        elif project:
             if not isinstance(project, Project):
                 project = get_project(project)
-            owner_url = project.url
-
-        if not owner_url:
+            self.move(project)
+        else:
             raise AttributeError("Can't set owner")
 
-        self.resource.patch({'owner': owner_url})
+    def move(self, project):
```

Comment:
I'm not sure if we want to have `Dataset.move`, when we already have `Project.move_here`, The location of a dataset is not its responsibility, this could make the code simpler and less entangled with `Project` code.

Reply:
My idea was to remove the project moving code (regardless of the function being a setter or a method) out of the Dataset class. Since the placing of datasets is a concern of projects and not of datasets, but between both options I much prefer the `move()` function as it is very explicit in the code that it is _moving_ the dataset to a particular location. A setter in this case wouldn't communicate the same semantics.